src/BloomFilter/BloomFilterAbstract.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace RocketLabs\BloomFilter;
+
+use RocketLabs\BloomFilter\Hash\HashInterface;
+use RocketLabs\BloomFilter\Persist\PersisterInterface;
+
+/**
+ * @author Igor Veremchuk igor.veremchuk@rocket-internet.de
+ */
+abstract class BloomFilterAbstract implements BloomFilterInterface
+{
+    /** @var int */
+    protected $bitSize;
+    /** @var int */
+    protected $hashCount;
+    /** @var PersisterInterface */
+    protected $persister;
+    /** @var HashInterface */
+    protected $hash;
+
+    /**
+     * @param PersisterInterface $persister
+     * @param HashInterface $hash
+     * @param int $setSize
+     * @param float $falsePositiveProbability
+     *
+     * @return BloomFilterInterface
+     */
+    public static function createFromSetSize(
+        PersisterInterface $persister,
+        HashInterface $hash,
+        $setSize,
+        $falsePositiveProbability = 0.001
+    ) {
+
+        $bloom = new static($persister, $hash);
+        $bloom->init($setSize, $falsePositiveProbability);
+
+        return $bloom;
+    }
+
+    /**
+     * @param PersisterInterface $persister
+     * @param HashInterface $hash
+     */
+    public function __construct(PersisterInterface $persister, HashInterface $hash)
```

Comment:
Would it not be better to have this method and the init with private visibility?